src/universal/data/notification.js

```diff
@@ -0,0 +1,50 @@
+// @flow
+
+/**
+ * Defines the js-land type of a Notification, and functions to operate on them.
+ */
+
+import {
+  ADD_TO_TEAM,
+  DENY_NEW_USER,
+  INVITEE_APPROVED,
+  KICKED_OUT,
+  PAYMENT_REJECTED,
+  PROJECT_INVOLVES,
+  PROMOTE_TO_BILLING_LEADER,
+  REQUEST_NEW_USER,
+  TEAM_ARCHIVED,
+  TEAM_INVITE
+} from '../utils/constants';
+
+type Notification = {
+  id: string,
+  orgId: string,
+  startAt: Date,
+  teamId?: string,
+  type:
+    | ADD_TO_TEAM
+    | DENY_NEW_USER
+    | INVITEE_APPROVED
+    | KICKED_OUT
+    | PAYMENT_REJECTED
+    | PROJECT_INVOLVES
+    | PROMOTE_TO_BILLING_LEADER
+    | REQUEST_NEW_USER
+    | TEAM_ARCHIVED
+    | TEAM_INVITE,
+  userIds: string[]
+};
+
+// Note: this is a conservative estimate; not all of these notifications "require" action,
+// but they all offer an action other than to clear the notification.
+const NOTIFICATION_TYPES_REQUIRING_ACTION = new Set([
```

Comment:
I think we should scope this down to: `PAYMENT_REJECTED`, `REQUEST_NEW_USER`, and `TEAM_INVITE` as the others can be clearable and the other action is options (see board, team, org).

Reply:
Updated the list, feel free to push back if you want, but the design intent in my mind was that when there was a clear button with another action button, the other action was optional, not required.

Reply:
That's perfect! I'm going to update the comment for accuracy.